File: common/src/main/java/com/pluxity/user/repository/RolePermissionRepository.java
package com.pluxity.user.repository;

import com.pluxity.permission.PermissionGroup;
import com.pluxity.user.entity.Role;
import com.pluxity.user.entity.RolePermission;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface RolePermissionRepository extends JpaRepository<RolePermission, Long> {
    void deleteAllByPermissionGroup(PermissionGroup permissionGroup);

    @Modifying
    @Query("DELETE FROM RolePermission rp WHERE rp.role = :role")
    void deleteAllByRole(@Param("role") Role role);
}

File: common/src/main/kotlin/com/pluxity/user/repository/RolePermissionRepository.kt
package com.pluxity.user.repository

import com.pluxity.permission.Permission
import com.pluxity.user.entity.Role
import com.pluxity.user.entity.RolePermission
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface RolePermissionRepository : JpaRepository<RolePermission, Long> {
    fun deleteAllByPermission(permission: Permission)

    @Modifying
    @Query("DELETE FROM RolePermission rp WHERE rp.role = :role")
    fun deleteAllByRole(
        @Param("role") role: Role,
    )
}
